doc/development.md

```diff
@@ -35,24 +35,46 @@
 
 ## Building Vdsm
 
-To configure sources (run `./configure --help` to see configuration options):
+To configure sources (create Makefiles and configure script): 
 
     git clean -xfd
     ./autogen.sh --system --enable-timestamp
-    make
 
-To test Vdsm (refer to tests/README for further tests information):
+(It is recommended to start fresh after an update, from the previous step, before
+building RPMs, because the generated Makefile also contains the release number.)
```

Comment:
I would keep this simple for the basic build flow. We can discuss rebasing in a section about preparing and updating prs.

Reply:
This document doesn't really talk about how to commit/rebase/push/merge things back to the upstream repo (yet). I assume that the process isn't that VDSM-specific, but rather should be the same for all of oVirt? So I'm not sure into how much detail we should go about that here.

Reply:
We needn't be that specific, this is not a git(hub) tutorial.  I'd just either drop the remark about release numbers or add a remark about making sure to be on the top of the upstream branch.  The point is that if you are behind the upstream branch, you may be on an old tag, resulting in building rpm's with an old version.

Reply:
I though the point was that even if you are up-to-date the Makefile would not necessarily be regenerated and could keep the old number. But now I see that it _does_ seem to be created new each time I run autogen.
So I don't really understand the point, because that you get old packages when you build an old version of the code seems kind of obvious. What, then, _is_ the purpose of `git clean -xfd`?

Reply:
> I though the point was that even if you are up-to-date the Makefile would not necessarily be regenerated and could keep the old number. But now I see that it _does_ seem to be created new each time I run autogen.
> So I don't really understand the point, because that you get old packages when you build an old version of the code seems kind of obvious. 

OK, then I would drop the change of this piece of the document.

> What, then, _is_ the purpose of `git clean -xfd`?

There might be some argument beyond "just cleanup to be sure", but GitHub doesn't make easy to find PR comments, so maybe Albert or Nir can clarify it.

Reply:
Running "git clean -dxf" is known to work for creating good build. One known issue
this solves is that the generated Makefile contains the RELEASE string
("{timestamp}.git{commit-hash}"). If you create a build with the same Makefile,
you always get the same release string with the wrong release and upgrading the 
vdsm does not work.

Reply:
I tested now with patch on current master, and running autogen.sh was enough to regenerate the makefile.

Before running autogen.sh:

```
$ grep 'RELEASE =' Makefile.before 
PACKAGE_BASE_RELEASE = 1
PACKAGE_RELEASE = 1
```

After running autogen.sh:

```
$ grep 'RELEASE =' Makefile
PACKAGE_BASE_RELEASE = 202206201318.gite763a6b19
PACKAGE_RELEASE = 202206201318.gite763a6b19
```

So we can remove the git clean from the instructions.

Reply:
I'm not sure what the deciding factor is, but in my tests the Makefile was not touched by running autogen.sh again.